pingpong/pingpong.go

```diff
@@ -127,26 +127,21 @@ func (pp *pingPonger) pong(ctx context.Context, to protocol.PeerAddress) error {
 	}
 }
 
-func (pp *pingPonger) propagatePing(ctx context.Context, body protocol.MessageBody) error {
-	// TODO: Wrap errors in custom error types.
-
+func (pp *pingPonger) propagatePing(ctx context.Context, sender protocol.PeerID, body protocol.MessageBody) error {
 	peerAddrs, err := pp.dht.PeerAddresses()
 	if err != nil {
 		return err
 	}
-	if peerAddrs == nil {
-		return fmt.Errorf("nil peer addresses")
-	}
-	if len(peerAddrs) <= 0 {
-		return fmt.Errorf("empty peer addresses")
-	}
 
 	// Using the messaging sending channel protects the pinger/ponger from
 	// cascading time outs, but will still capture back pressure
-	for i := range peerAddrs {
+	for _, addr := range peerAddrs {
```

Comment:
You probably want to use the same `phi.ParForAll(numWorkers, ...)` idea that has been suggested for `broadcaster` and `multicaster`.